server/mongo/src/storage.ts

```diff
@@ -195,18 +190,18 @@ class MongoAdapter extends MongoAdapterBase {
         return await this.db.collection(domain).bulkWrite(ops as any)
       } else {
         if (tx.retrieve === true) {
-          const result = await this.db.collection(domain).findOneAndUpdate({ _id: tx.objectId }, operations, { returnDocument: 'after' })
+          const result = await this.db.collection(domain).findOneAndUpdate({ _id: tx.objectId }, tx.operations, { returnDocument: 'after' })
```

Comment:
We need $set: {
modifiedBy: tx.modifiedBy,
modifiedOn: tx.modifiedOn
}

To be added.